webapp/channels/src/components/suggestion/generic_user_provider.tsx

```diff
@@ -13,85 +13,72 @@ import {isGuest} from 'mattermost-redux/utils/user_utils';
-        }
-
-        const username = item.username;
-        let description = '';
-
-        if ((item.first_name || item.last_name) && item.nickname) {
-            description = `- ${Utils.getFullName(item)} (${item.nickname})`;
-        } else if (item.nickname) {
-            description = `- (${item.nickname})`;
-        } else if (item.first_name || item.last_name) {
-            description = `- ${Utils.getFullName(item)}`;
-        }
-
-        return (
-            <div
-                className={className}
-                onClick={this.handleClick}
-                onMouseMove={this.handleMouseMove}
-                {...Suggestion.baseProps}
-            >
-                <Avatar
-                    size='xxs'
-                    username={username}
-                    url={Client4.getUsersRoute() + '/' + item.id + '/image?_=' + (item.last_picture_update || 0)}
-                />
-                <div className='suggestion-list__ellipsis'>
-                    <span className='suggestion-list__main'>
-                        {'@' + username}
-                    </span>
-                    {description}
-                </div>
-                {item.is_bot && <BotTag/>}
-                {isGuest(item.roles) && <GuestTag/>}
-            </div>
-        );
+    const username = item.username;
+    let description = '';
+
+    if ((item.first_name || item.last_name) && item.nickname) {
+        description = `- ${Utils.getFullName(item)} (${item.nickname})`;
+    } else if (item.nickname) {
+        description = `- (${item.nickname})`;
+    } else if (item.first_name || item.last_name) {
+        description = `- ${Utils.getFullName(item)}`;
     }
-}
 
-export default class UserProvider extends Provider {
+    return (
+        <SuggestionContainer
+            ref={ref}
+            {...props}
+        >
+            <Avatar
+                size='xxs'
+                username={username}
+                url={Client4.getUsersRoute() + '/' + item.id + '/image?_=' + (item.last_picture_update || 0)}
+            />
+            <div className='suggestion-list__ellipsis'>
+                <span className='suggestion-list__main'>
+                    {'@' + username}
+                </span>
+                {description}
+            </div>
+            {item.is_bot && <BotTag/>}
+            {isGuest(item.roles) && <GuestTag/>}
+        </SuggestionContainer>
+    );
+});
+GenericUserSuggestion.displayName = 'GenericUserSuggestion';
+
+export default class GenericUserProvider extends Provider {
     autocompleteUsers: (text: string) => Promise<UserAutocomplete>;
+
     constructor(searchUsersFunc: (username: string) => Promise<UserAutocomplete>) {
         super();
         this.autocompleteUsers = searchUsersFunc;
     }
-    async handlePretextChanged(pretext: string, resultsCallback: (res: ProviderResults) => void) {
+
+    handlePretextChanged(pretext: string, resultsCallback: ResultsCallback<UserProfile>) {
```

Comment:
This function shouldn't have been async to begin with, but it didn't cause an error until I made `Provider` into a proper abstract class, so we never noticed before. This behaviour should be the same as before because the previous version returned a promise which would've been interpreted as truthy by the `SuggestionBox` which calls this method